cmd/erasure-server-pool.go

```diff
@@ -292,46 +292,17 @@ func (z *erasureServerPools) getPoolIdxExisting(ctx context.Context, bucket, obj
 // getPoolIdx returns the found previous object and its corresponding pool idx,
 // if none are found falls back to most available space pool.
 func (z *erasureServerPools) getPoolIdx(ctx context.Context, bucket, object string, size int64) (idx int, err error) {
-	if z.SinglePool() {
-		return 0, nil
-	}
-
-	errs := make([]error, len(z.serverPools))
-	objInfos := make([]ObjectInfo, len(z.serverPools))
-
-	var wg sync.WaitGroup
-	for i, pool := range z.serverPools {
-		wg.Add(1)
-		go func(i int, pool *erasureSets) {
-			defer wg.Done()
-			objInfos[i], errs[i] = pool.GetObjectInfo(ctx, bucket, object, ObjectOptions{})
-		}(i, pool)
-	}
-	wg.Wait()
-
-	for i, err := range errs {
-		if err != nil && !isErrObjectNotFound(err) {
-			return -1, err
-		}
-		if isErrObjectNotFound(err) {
-			// No object exists or its a delete marker,
-			// check objInfo to confirm.
-			if objInfos[i].DeleteMarker && objInfos[i].Name != "" {
-				return i, nil
-			}
-			// objInfo is not valid, truly the object doesn't
-			// exist proceed to next pool.
-			continue
-		}
-		// object exists at this pool.
-		return i, nil
+	idx, err = z.getPoolIdxExisting(ctx, bucket, object)
+	if err != nil && !isErrObjectNotFound(err) {
```

Comment:
Missing `return idx, nil` here ?

Reply:
Yes @vadmeste that is the bug I have been looking for.